packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { LibraryCommandletExecutor } from '@salesforce/salesforcedx-utils-vscode/out/src';
+import { notificationService } from '@salesforce/salesforcedx-utils-vscode/out/src/commands';
+import { CancelResponse, ContinueResponse, ParametersGatherer } from '@salesforce/salesforcedx-utils-vscode/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { format } from 'util';
+import * as vscode from 'vscode';
+import { OUTPUT_CHANNEL } from '../channels';
+import { nls } from '../messages';
+import { SfdxCommandlet, SfdxWorkspaceChecker } from './util';
+
+const RENAME_LIGHTNING_COMPONENT_EXECUTOR = 'force_rename_lightning_component';
+const RENAME_INPUT_PLACEHOLDER = 'rename_component_input_placeholder';
+const RENAME_INPUT_PROMPT = 'rename_component_input_prompt';
+const RENAME_INPUT_DUP_ERROR = 'rename_component_input_dup_error';
+const RENAME_WARNING = 'rename_component_warning';
+const LWC = 'lwc';
+const AURA = 'aura';
+
+export class RenameLwcComponentExecutor extends LibraryCommandletExecutor<ComponentName> {
+  private sourceFsPath: string;
+  constructor(sourceFsPath: string) {
+    super(
+      nls.localize(RENAME_LIGHTNING_COMPONENT_EXECUTOR),
+      RENAME_LIGHTNING_COMPONENT_EXECUTOR,
+      OUTPUT_CHANNEL
+    );
+    this.sourceFsPath = sourceFsPath;
+  }
+
+  public async run(
+    response: ContinueResponse<ComponentName>
+    ): Promise<boolean> {
+      const newComponentName = response.data.name;
+      if (newComponentName && this.sourceFsPath) {
+        await renameComponent(this.sourceFsPath, newComponentName);
+        return true;
+      }
+      return false;
```

Comment:
swinging back around to the telemetry.  We do get a ping about this command being called b/c of the LibraryCommandletExecutor extends...however we don't know if we actually do it as determined by this logic.  It's fine to push that off for now as we need a plan, but I was just calling it out previously.